roomserver/internal/input_membership.go

```diff
@@ -106,6 +107,13 @@ func updateMembership(
 		return updates, nil
 	}
 
+	if add == nil {
```

Comment:
Looking at the comments, `add==nil` if its a leave, as:

```
// Default the membership to Leave if no event was added or removed.
oldMembership := gomatrixserverlib.Leave
newMembership := gomatrixserverlib.Leave
```

Reply:
But in that case it should be that the old membership is whatever it was before and the new membership is now `leave`. If it's old `leave` and new `leave` then we should have filtered that out in the check above it.